test/expected/build/library.j/dest/resources/library/j/some.js

```diff
@@ -1,16 +1,84 @@
 /*!
  * ${copyright}
  */
+sap.ui.define([
+	"sap/ui/core/Control", 
+	"./dependency",
+	"./dependency2"
+], (Control, { MyValidEnum }, library) => {
 
-sap.ui.define([],
-	function() {
-	"use strict";
+	const { AnotherValidEnum } = library;
+	const { Buzz } = AnotherValidEnum;
+	const { AnotherValidEnum: Zzz } = library;
+	const { TitleLevel } = sap.ui.core;
 
 	/**
-	 * @alias library.j
-	 * @namespace
+	 * @class
+	 * My super documentation of this class
+	 *
+	 * @extends sap.ui.core.Control
+	 *
+	 * @author SAP SE
+	 * @version ${version}
+	 *
 	 * @public
+	 * @alias testlib.ValidPropertyDefaultValue
+	 * @ui5-metamodel text
 	 */
-	var SomeFunction = function() {};
+	var ValidPropertyDefaultValue = Control.extend(
+		"testlib.ValidPropertyDefaultValue",
+		{
+			metadata: {
+				properties: {
+					/**
+					 * validPropertyDefaultValueEnumSimpleDestructuring
+					 */
+					validPropertyDefaultValueEnumSimpleDestructuring: {
+						type: "testlib.AnotherValidEnum",
+						group: "Misc",
+						defaultValue: AnotherValidEnum.Buzz,
+					},
 
+					/**
+					 * validPropertyDefaultValueEnumChainedDestructuring
+					 */
+					validPropertyDefaultValueEnumChainedDestructuring: {
+						type: "testlib.AnotherValidEnum",
+						group: "Misc",
+						defaultValue: Buzz,
+					},
+
+					/**
+					 * validPropertyDefaultValueEnumDestructuringWithRename
+					 */
+					validPropertyDefaultValueEnumDestructuringWithRename: {
+						type: "testlib.AnotherValidEnum",
+						group: "Misc",
+						defaultValue: Zzz.Fizz,
+					},
+
+					/**
+					 * validPropertyDefaultValueEnumViaDestructuringInArrowFn
+					 */
+					validPropertyDefaultValueEnumViaDestructuringInArrowFn: {
+						type: "testlib.MyValidEnum",
+						group: "Misc",
+						defaultValue: MyValidEnum.Foo,
+					},
+
+					/**
+					 * validPropertyDefaultValueEnumViaDestructuringGlobal
+					 */
+					validPropertyDefaultValueEnumViaDestructuringGlobal: {
+						type: "sap.ui.core",
+						group: "Misc",
+						defaultValue: TitleLevel,
```

Comment:
... and also the default value doesn't make sense. Or is this a test for some invalid use case?

Reply:
Nope, it was a mistake. Now should be the correct one